django/db/backends/mysql/schema.py

```diff
@@ -32,8 +32,10 @@ def quote_value(self, value):
     def skip_default(self, field):
         """
         MySQL doesn't accept default values for some data types and implicitly
-        treats these columns as nullable.
+        treats these columns as nullable. MariaDB 10.2+ fixes this.
         """
+        if self.connection.mysql_is_mariadb and self.connection.mysql_version[:2] > (10, 2):
+            return False
```

Comment:
I think adding a MariaDB backend would be a good idea eventually but first I'd like to get Django using `pymysql` as well, which I have half-done at https://github.com/adamchainz/django-pymysql-backend . But then we'd possibly have 2 MySQL backends and 2 MariaDB backends inheriting from them 😞 .